### DISCUSSION TITLE
Can nczarr consolidate the metadata (create .zmetadata) ?

### DISCUSSION THREAD

**Discussion by Alexander-Barth** [Unidata/netcdf-c#2832] [General]
If I create a zarr file with NetCDF 4.9.2 (using `file:///tmp/test3.zarr#mode=zarr`), xarray (2023.12.0) gives the following warning:

```python
import xarray as xr
ds = xr.open_dataset("/tmp/test3.zarr",engine = "zarr")
# warning:
<ipython-input-7-ebe2b0c6ab00>:1: RuntimeWarning: Failed to open Zarr store with consolidated metadata, but successfully read with non-consolidated metadata. This is typically much slower for opening a dataset. To silence this warning, consider:
1. Consolidating metadata in this existing store with zarr.consolidate_metadata().
2. Explicitly setting consolidated=False, to avoid trying to read consolidate metadata, or
3. Explicitly setting consolidated=True, to raise an error in this case instead of falling back to try reading non-consolidated metadata.
```

Rather than setting `consolidated=False`, I am wondering if it is possible to generate directly consolidated files from nczarr.

**Comment by DennisHeimbigner** [comment]
NCZarr does not yet support consolidated meta-data. The problem is that I can find no standard for
defining it. When there is a standard, then I plan to support it.

**Comment by Alexander-Barth** [comment]
Indeed, it is surprising that `.zmetadata` it is not mentioned here:
https://github.com/zarr-developers/zarr-python/blob/main/docs/spec/v2.rst
https://portal.ogc.org/files/100727

**Comment by DennisHeimbigner** [comment]
There may be a PR for the v2 spec that defines it.
I have not checked in a while.
